sys/kern/kern_proc.c

```diff
@@ -2574,8 +2579,8 @@ proc_read_string_properly(struct thread *td, struct proc *p,
 	if (!cheri_can_access(sptr, CHERI_PERM_LOAD,
 	    (__cheri_addr ptraddr_t)sptr, valid))
 		return (EPROT);
-	readlen = proc_readmem(td, p, (__cheri_addr ptraddr_t)sptr, buf, valid);
-	if (readlen <= 0)
+	if (proc_readmem(td, p, (__cheri_addr ptraddr_t)sptr, buf, valid) !=
```

Comment:
Hmmm, I think this is on purpose in case a string is shorter than a capability bounds and you cross into an unmapped page after the nul terminator.